src/lib/unittest/__init__.py

```diff
@@ -186,26 +240,13 @@ def appendResult(self,res,actual,expected,feedback):
             print(msg)
             self.assertFailed += 1
 
-    def assertRaises(self, exception, callable=None, *args, **kwds):
-        # with is currently not supported hence we just try and catch
-        if callable is None:
-            raise NotImplementedError("assertRaises does currently not support assert contexts")
-        if kwds:
-            raise NotImplementedError("assertRaises does currently not support **kwds")
-
-        res = False
-        actualerror = str(exception())
+    def assertRaises(self, expected_exception, *args, **kwargs):
+        context = _AssertRaisesContext(expected_exception, self)
         try:
-            callable(*args)
-        except exception as ex:
-            res = True
-        except Exception as inst:
-            actualerror = str(inst)
-            print("ACT = ", actualerror, str(exception()))
-        else:
-            actualerror = "No Error"
-
-        self.appendResult(res, str(exception()), actualerror, "")
+            return context.handle(args, kwargs)
+        finally:
+            # bpo-23890: manually break a reference cycle
+            context = None
```

Comment:
This is an odd-shaped invocation. Are you copying code from Python mainline, and if so where/which version? (I'm guessing you are, because reference cycles aren't a thing with a proper GC like Javascript has)

Reply:
yeah - the code is from https://github.com/python/cpython/blob/3.7/Lib/unittest/case.py#L757


There's also a similar invocation in the `_AssertRaisesContext `
```
        finally:
            # bpo-23890: manually break a reference cycle
            self = None
```
https://github.com/python/cpython/blob/3.7/Lib/unittest/case.py#L179

I wasn't sure about them - so I just left them in...